src/Controller/AppController.php

```diff
@@ -373,4 +373,40 @@ protected function getObjectNav($id): array
 
         return (array)Hash::get($objectNav, sprintf('%s.%s', $objectNavModule, $id), []);
     }
+
+    /**
+     * Load tree data.
+     *
+     * @return void
+     */
+    public function treeJson(): void
+    {
+        $this->request->allowMethod(['get']);
+        $query = $this->Modules->prepareQuery($this->request->getQueryParams());
+        $root = Hash::get($query, 'root');
+        $full = Hash::get($query, 'full');
+        unset($query['full']);
+        unset($query['root']);
+        try {
+            if (empty($full)) {
+                $key = (empty($root)) ? 'roots' : 'parent';
+                $val = (empty($root)) ? '' : $root;
+                if (empty($query['filter'])) {
+                    $query['filter'] = [];
+                }
+                $query['filter'][$key] = $val;
+            }
+            $response = $this->apiClient->getObjects('folders', $query);
+        } catch (BEditaClientException $error) {
+            $this->log($error, LogLevel::ERROR);
```

Comment:
also missing import for `LogLevel` 

but we can safely use ` $this->log($error, 'error');` avoiding import